core/sdk-core/src/main/java/software/amazon/awssdk/core/compression/CompressionType.java

```diff
@@ -15,60 +15,102 @@
-        CompressionType.class, CompressionType::toString);
+    private String id;
 
-    private final String value;
+    private CompressionType(String id) {
+        this.id = id;
+    }
 
-    CompressionType(String value) {
-        this.value = value;
+    /**
+     * Creates a new {@link CompressionType} of the given value.
+     */
+    public static CompressionType of(String value) {
+        Validate.paramNotBlank(value, "compressionType");
+        return CompressionTypeCache.put(value);
     }
 
-    @Override
-    public String toString() {
-        return String.valueOf(value);
+    /**
+     * Returns the {@link Set} of {@link String}s of compression types supported by the SDK.
+     */
+    public static Set<String> compressionTypes() {
+        return compressorMap.keySet();
     }
 
     /**
-     * Use this in place of valueOf to convert the raw string into the enum value.
-     *
-     * @param value
-     *        real value
-     * @return SupportedEncodings corresponding to the value
+     * Whether or not the compression type is supported by the SDK.
      */
-    public static CompressionType fromValue(String value) {
-        if (value == null) {
-            return null;
-        }
-        return VALUE_MAP.getOrDefault(value, UNKNOWN_TO_SDK_VERSION);
+    public static boolean isSupported(String compressionType) {
+        return compressionTypes().contains(compressionType);
     }
 
     /**
-     * Use this in place of {@link #values()} to return a {@link Set} of all values known to the SDK. This will return
-     * all known enum values except {@link #UNKNOWN_TO_SDK_VERSION}.
-     *
-     * @return a {@link Set} of known {@link CompressionType}s
+     * Maps the {@link CompressionType} to its corresponding {@link Compressor}.
      */
-    public static Set<CompressionType> knownValues() {
-        Set<CompressionType> knownValues = EnumSet.allOf(CompressionType.class);
-        knownValues.remove(UNKNOWN_TO_SDK_VERSION);
-        return knownValues;
+    public Compressor newCompressor() {
+        Compressor compressor = compressorMap.getOrDefault(this.id, null);
+        if (compressor == null) {
+            throw new UnsupportedOperationException("The compression type " + id + " does not have an implementation of "
+                                                    + "Compressor");
+        }
+        return compressor;
+    }
+
+    @Override
+    public String toString() {
+        return id;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        CompressionType that = (CompressionType) o;
+        return Objects.equals(id, that.id)
+            && Objects.equals(compressorMap, that.compressorMap);
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
We can improve the Coverage of newly added classes by performing EqualsVerifier test , this will make sure equals and hashcode are tested

import nl.jqno.equalsverifier.EqualsVerifier;

Reply:
added tests for CompressionType and RequestCompressionConfiguration